internal/model/puller.go

```diff
@@ -658,12 +649,9 @@ nextFile:
 					return false
 				}
 
-				// Only done on second to last puller attempt
-				if checksum {
-					hasher.Write(buf)
-					hash := hasher.Sum(nil)
-					hasher.Reset()
-					if !bytes.Equal(hash, block.Hash) {
+				hash, err := scanner.VerifyBuffer(buf, block)
+				if err != nil && hash != nil {
+					if hash != nil {
```

Comment:
Inner if can never be false, and I don't think we ever return err==nil, hash==nil right?